modules/prompt/functions/prompt_sorin_setup

```diff
@@ -32,29 +32,25 @@
 # Load dependencies.
 pmodload 'helper'
 
-function prompt_sorin_async_callback {
-  case $1 in
-    prompt_sorin_async_git)
-      # We can safely split on ':' because it isn't allowed in ref names.
-      IFS=':' read _git_target _git_post_target <<<"$3"
-
-      # The target actually contains 3 space separated possibilities, so we need to
-      # make sure we grab the first one.
-      _git_target=$(coalesce ${(@)${(z)_git_target}})
-
-      if [[ -z "$_git_target" ]]; then
-        # No git target detected, flush the git fragment and redisplay the prompt.
-        if [[ -n "$_prompt_sorin_git" ]]; then
-          _prompt_sorin_git=''
-          zle && zle reset-prompt
-        fi
-      else
-        # Git target detected, update the git fragment and redisplay the prompt.
-        _prompt_sorin_git="${_git_target}${_git_post_target}"
-        zle && zle reset-prompt
```

Comment:
So are we already using both `zle` _and_ `zsh-async`? It does seem weird that we'd be using two async frameworks...

Reply:
zle is a zsh built-in and does a ton of stuff related to line editing. For some reason there's also some functionality in there around managing file descriptors for sub-shells.